src/realm/object-store/sync/app_backing_store.hpp

```diff
@@ -0,0 +1,143 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef REALM_OS_APP_BACKING_STORE_HPP
+#define REALM_OS_APP_BACKING_STORE_HPP
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <realm/util/function_ref.hpp>
+
+namespace realm {
+
+class SyncAppMetadata;
+class SyncMetadataManager;
+class SyncUser;
+
+namespace app {
+
+class App;
+
+struct BackingStoreConfig {
```

Comment:
All of these config settings are specific to RealmBackingStore and don't make sense as part of the virtual interface.